File: uhrwerk-core/src/main/java/io/qimia/uhrwerk/metastore/model/data/PartitionDependency.java
package io.qimia.uhrwerk.metastore.model.data;


public class PartitionDependency {

  private long id;
  private long partitionId;
  private long dependencyPartitionId;
  private java.sql.Timestamp createdTs;
  private java.sql.Timestamp updatedTs;
  private String description;


  public long getId() {
    return id;
  }

  public void setId(long id) {
    this.id = id;
  }


  public long getPartitionId() {
    return partitionId;
  }

  public void setPartitionId(long partitionId) {
    this.partitionId = partitionId;
  }


  public long getDependencyPartitionId() {
    return dependencyPartitionId;
  }

  public void setDependencyPartitionId(long dependencyPartitionId) {
    this.dependencyPartitionId = dependencyPartitionId;
  }


  public java.sql.Timestamp getCreatedTs() {
    return createdTs;
  }

  public void setCreatedTs(java.sql.Timestamp createdTs) {
    this.createdTs = createdTs;
  }


  public java.sql.Timestamp getUpdatedTs() {
    return updatedTs;
  }

  public void setUpdatedTs(java.sql.Timestamp updatedTs) {
    this.updatedTs = updatedTs;
  }


  public String getDescription() {
    return description;
  }

  public void setDescription(String description) {
    this.description = description;
  }

}

File: uhrwerk-metastore/src/main/kotlin/io/qimia/uhrwerk/common/metastore/model/PartitionDependency.kt
package io.qimia.uhrwerk.common.metastore.model

import java.sql.Timestamp
import java.time.LocalDateTime

data class PartitionDependency(
    var id: Long? = null,
    var partitionId: Long? = null,
    var dependencyPartitionId: Long? = null,
    var description: String? = null,
    var deactivatedTs: LocalDateTime? = null,
    var createdTs: Timestamp? = null,
    var updatedTs: Timestamp? = null
) : BaseModel {
    override fun id(id: Long?) {
        this.id = id
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is PartitionDependency) return false
        if (id != null && other.id != null) if (id != other.id) return false
        if (partitionId != other.partitionId) return false
        if (dependencyPartitionId != other.dependencyPartitionId) return false
        if (description != other.description) return false
        if (deactivatedTs != other.deactivatedTs) return false
        if (createdTs != other.createdTs) return false
        if (updatedTs != other.updatedTs) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id?.hashCode() ?: 0
        result = 31 * result + (partitionId?.hashCode() ?: 0)
        result = 31 * result + (dependencyPartitionId?.hashCode() ?: 0)
        result = 31 * result + (description?.hashCode() ?: 0)
        result = 31 * result + (deactivatedTs?.hashCode() ?: 0)
        result = 31 * result + (createdTs?.hashCode() ?: 0)
        result = 31 * result + (updatedTs?.hashCode() ?: 0)
        return result
    }
}